src/server/lib/okta/register.ts

```diff
@@ -0,0 +1,64 @@
+import { Status, User } from '@/server/models/okta/User';
+import {
+  createUser,
+  fetchUser,
+  activateUser,
+  reactivateUser,
+} from '@/server/lib/okta/api/users';
+import {
+  OktaError,
+  ResourceAlreadyExistsError,
+} from '@/server/models/okta/Error';
+
+export const register = async (email: string): Promise<User> => {
+  try {
+    return await createUser(email);
+  } catch (error) {
+    if (error instanceof ResourceAlreadyExistsError) {
+      const user: User = await fetchUser({ id: email });
+      const { status } = user;
+      switch (status) {
+        case Status.STAGED: {
+          await activateUser({ id: user.id });
+          return user;
+        }
+        case Status.PROVISIONED: {
+          await reactivateUser({ id: user.id });
+          return user;
+        }
+        case Status.ACTIVE || Status.RECOVERY || Status.PASSWORD_EXPIRED: {
+          // TODO: implement reset password email; this throws an error for now so registration fails
+          throw new OktaError(
+            `Okta registration failed with unaccepted Okta user status: ${user.status}`,
+          );
+        }
+        default:
+          throw new OktaError(
+            `Okta registration failed with unaccepted Okta user status: ${user.status}`,
+          );
+      }
+    } else {
+      throw error;
+    }
+  }
+};
+
```

Comment:
Is it worth explaining in comments why the user could have different status? The broader question is what is the best way to document the Okta registration integration.

Reply:
I think it would be best to put this information in `/docs` along with the user state diagram when it's ready. It's still a bit fluid for now while we're finalising the forgot password flow